packages/transformers/src/lit-transformer.ts

```diff
@@ -0,0 +1,263 @@
+
+interface ClassDecoratorTransformer {
+  kind: 'classDecorator';
+  decoratorName: string;
+
+  visit(
+    mutations: LitElementMutations,
+    class_: ts.ClassDeclaration,
+    decorator: ts.Decorator
+  ): void;
+}
+
+interface MemberDecoratorTransformer {
+  kind: 'memberDecorator';
+  decoratorName: string;
+
+  visit(
+    mutations: LitElementMutations,
+    member: ts.ClassElement,
+    decorator: ts.Decorator
+  ): void;
+}
+
+const unreachable = (x: never) => x;
+
+/**
+ * Configurable transformer for LitElement classes.
+ */
+export class LitTransformer {
+  private _context: ts.TransformationContext;
+
+  private _classDecoratorVisitors = new Map<
+    string,
+    ClassDecoratorTransformer[]
+  >();
+
+  private _memberDecoratorVisitors = new Map<
+    string,
+    MemberDecoratorTransformer[]
+  >();
+
+  private _removeNodes = new Set<ts.Node>();
+
+  constructor(context: ts.TransformationContext, visitors: Array<Visitor>) {
+    this._context = context;
+    for (const visitor of visitors) {
+      switch (visitor.kind) {
+        case 'classDecorator': {
+          let arr = this._classDecoratorVisitors.get(visitor.decoratorName);
+          if (arr === undefined) {
+            arr = [];
+            this._classDecoratorVisitors.set(visitor.decoratorName, arr);
+          }
+          arr.push(visitor);
+          break;
+        }
+
+        case 'memberDecorator': {
+          let arr = this._memberDecoratorVisitors.get(visitor.decoratorName);
+          if (arr === undefined) {
+            arr = [];
+            this._memberDecoratorVisitors.set(visitor.decoratorName, arr);
+          }
+          arr.push(visitor);
+          break;
+        }
+
+        default:
+          throw new Error(
+            `Internal error: unknown visitor kind ${
+              (unreachable(visitor) as Visitor).kind
+            }`
+          );
+      }
+    }
+  }
+
+  visit = (node: ts.Node): ts.VisitResult<ts.Node> => {
+    if (this._removeNodes.delete(node)) {
+      return undefined;
+    }
+    if (ts.isImportDeclaration(node)) {
+      return this._visitImportDeclaration(node);
+    }
+    if (ts.isClassDeclaration(node)) {
+      return this._visitClassDeclaration(node);
+    }
+    return ts.visitEachChild(node, this.visit, this._context);
+  };
+
+  private _visitImportDeclaration(node: ts.ImportDeclaration) {
+    if (
+      ts.isStringLiteral(node.moduleSpecifier) &&
+      node.moduleSpecifier.text === 'lit/decorators.js'
+    ) {
+      return undefined;
+    }
+    return node;
+  }
+
+  private _visitClassDeclaration(class_: ts.ClassDeclaration) {
+    const mutations = new LitElementMutations();
+
+    // Class decorators
+    for (const decorator of class_.decorators ?? []) {
+      if (
+        !ts.isCallExpression(decorator.expression) ||
+        !ts.isIdentifier(decorator.expression.expression)
+      ) {
+        continue;
+      }
+      const decoratorName = decorator.expression.expression.getText();
```

Comment:
Avoid calling getText(), as it's present on every node but isn't generally what you want (and it can throw for artificial nodes). We actually added a linter check to prevent calling getText in grimlock

We know that decorator.expression.expression is an Identifier so we can just get its text with the `text` property

Reply:
Done, removed all usage.